src/main/scala/TraceIO.scala

```diff
@@ -18,28 +18,33 @@ import freechips.rocketchip.diplomacy.{BundleBridgeSource, BundleBroadcast, Bund
 // used to connect the TracerV or Dromajo bridges (in FireSim and normal sim)
 //***************************************************************************
 
-case class TracedInstructionWidths(iaddr: Int, insn: Int, wdata: Int, cause: Int, tval: Int)
+case class TracedInstructionWidths(iaddr: Int, insn: Int, wdata: Option[Int], cause: Int, tval: Int)
 
 object TracedInstructionWidths {
-  def apply(tI: ExtendedTracedInstruction): TracedInstructionWidths =
-    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, tI.wdata.getWidth, tI.cause.getWidth, tI.tval.getWidth)
+  def apply(tI: ExtendedTracedInstruction): TracedInstructionWidths = {
+    val wdataWidth = tI.wdata match {
+      case Some(wd) => Some(wd.getWidth)
+      case None => None
+    }
+    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, wdataWidth, tI.cause.getWidth, tI.tval.getWidth)
+  }
 
-  // note: the wdata is not 0 here since we can't deal with 0 width wires
   def apply(tI: TracedInstruction): TracedInstructionWidths =
-    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, 1, tI.cause.getWidth, tI.tval.getWidth)
+    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, None, tI.cause.getWidth, tI.tval.getWidth)
 }
 
-class ExtendedTracedInstruction(implicit p: Parameters) extends TracedInstruction {
-  val wdata = UInt(xLen.W)
+class ExtendedTracedInstruction(extended: Boolean = true)(implicit p: Parameters) extends TracedInstruction {
+  val wdata = if (extended) Some(UInt(xLen.W)) else None
+
+  override def cloneType: this.type = new ExtendedTracedInstruction(extended).asInstanceOf[this.type]
```

Comment:
You don't have to change this, but you shouldn't need to define the cloneType if you make the parameters of the class public vals.